btcec/go.mod

```diff
@@ -1,16 +1,19 @@
 module github.com/btcsuite/btcd/btcec/v2
 
-go 1.17
+go 1.22.6
```

Comment:
I think we might need to keep this at an older version? Or at least have the old 2-part format, since this seems to break the unit tests (or the linter).

Reply:
i think there's something wrong with our CI here, still digging but the tests are failing locally but not on github actions hmmm

Reply:
turns out that locally all the sub-modules are pointed to the local versions as indicated by the `go.work`, which is now fixed once the versions are updated here.

Reply:
Hmm, I do know there's this hardware wallet (?) that relies on GopherJS (Go transpiler) which doesn't support the latest version. In the past, we wanted to introduce some code using generics, but backed off at their request. Though ultimately we can't hold it all back forever...

Reply:
hmm yeah didn't know that before. It looks like [they only support up to 0.19.0](https://github.com/gopherjs/gopherjs?tab=readme-ov-file#whats-new), should we use `0.19.0` instead?